### DISCUSSION TITLE
How should F# bootstrapping be handled in source-build?

### DISCUSSION THREAD

**Discussion by crummel** [dotnet/fsharp#13470] [General]
As I understand it the `Proto` build in F# bootstraps F# and requires a version of FSharp.Core [that doesn't participate in version flow](https://github.com/dotnet/fsharp/blob/539de8504bedaf51985ff5992b89b17d31b89618/eng/Versions.props#L42).  This brings in the FSharp.Core version used as a source-build prebuilt.  Should this package be updated with the previously-source-built version flow?  Is bootstrapping necessary in the source-build F# build?

**Comment by vzarytovskii** [comment]
I'll leave it for @KevinRansom to describe :)

**Comment by KevinRansom** [comment]
@vzarytovskii , thanks mate, I appreciate it :-)
@crummel.

We use a last known good to build the proto compiler, including a proto fsharp.core and a proto fsharp.build.dll, as it were.

There is a crappy hack in there to use a published fsharp.core, solely because we expect the fsharp.build.dll to load into visual studio, and we "need" to ensure that when we build the proto fsharp.build.dll it matches a published fsharp.core.

In source build that is absolutely not necessary ... and so you can change the condition 
to exclude doing this package reference on source build too.
Here: https://github.com/dotnet/fsharp/blob/2211866bd809e5160f3afe62257ced1de4ff358c/src/FSharp.Build/FSharp.Build.fsproj#L55
and here: https://github.com/dotnet/fsharp/blob/2211866bd809e5160f3afe62257ced1de4ff358c/src/FSharp.Build/FSharp.Build.fsproj#L63

**Comment by dsyme** [comment]
Converted to discussion
